core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -208,8 +211,8 @@ public InclusiveManifestEvaluator load(Integer specId) {
         TableProperties.SPLIT_LOOKBACK, TableProperties.SPLIT_LOOKBACK_DEFAULT);
 
     return CloseableIterable.transform(
-        CloseableIterable.wrap(planFiles(), files ->
-            new BinPacking.PackingIterable<>(files, splitSize, lookback, FileScanTask::length)),
+        CloseableIterable.wrap(planSplits(splitSize), splits ->
```

Comment:
What if we have a non-splittable file format?

Reply:
That's a good point. I'll see how can we incorporate that.

Reply:
+1 for having a check for splittable formats.